diff/autofd/autofd.go

```diff
@@ -22,9 +22,11 @@ type Func struct {
 	Deriv string // Name of the output derivative function.
 }
 
-// Derivative generates code for both the first and second derivatives from the given function declaration.
-func Derivative(w io.Writer, f Func) error {
-	gen, err := newGenerator(w, f)
+// Derivative generates code for derivatives from the given function declaration.
+// It generates only the first derivative when d2 is false and generates
+// both first and second derivatives otherwise.
```

Comment:
```
// If d2 is true, both the generated function returns both the first and second
// derivatives. Otherwise only the first derivative function is generated.
```